src/array-repeat-strategy.js

```diff
@@ -20,16 +20,82 @@ export class ArrayRepeatStrategy {
   * @param items The new array instance.
   */
   instanceChanged(repeat, items) {
-    if (repeat.viewsRequireLifecycle) {
-      let removePromise = repeat.removeAllViews(true);
-      if (removePromise instanceof Promise) {
-        removePromise.then(() => this._standardProcessInstanceChanged(repeat, items));
+    const itemNameInBindingContext = repeat.local;
+    
+    const children = repeat.viewSlot.children;
+    let itemsPreviouslyInViews = children.map(view => view.bindingContext[itemNameInBindingContext]);
+    
+    let itemsLength = items.length;
+    
+    // remove items no longer in the instance
+    if (itemsPreviouslyInViews.length > 0) {
+      let indiciesToRemove = [];
+      
+      itemsPreviouslyInViews.forEach((oldItem, index) => {
```

Comment:
Please replace the `.forEach` with a regular for-loop.  Will reduce the memory pressure a bunch by eliminating all the closures that are created.